From 205528dd3813394f9c5c3f3f38d637c9f03102f5 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Sat, 9 Feb 2019 18:38:07 +0100
Subject: [PATCH 3/3] Revert "Remove udev and bash-completion installation from
 Makefile"

It is entirely unclear why was this removed, but it was usueful for
distribution package maintainers. Bring it back.

This reverts commit 991615a2d715c2a84d0f5f9a0bbe1ec697c6ee91.
---
 Makefile | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/Makefile b/Makefile
index 2514cc8..acbe657 100644
--- a/Makefile
+++ b/Makefile
@@ -48,6 +48,12 @@ DIST_DIR = $(MINIPRO)-$(GIT_TAG)
 BIN_INSTDIR=$(DESTDIR)$(PREFIX)/bin
 MAN_INSTDIR=$(DESTDIR)$(PREFIX)/share/man/man1
 
+UDEV_DIR=$(shell pkg-config --define-variable=prefix=$(PREFIX) --silence-errors --variable=udevdir udev)
+UDEV_RULES_INSTDIR=$(DESTDIR)$(UDEV_DIR)/rules.d
+
+COMPLETIONS_DIR=$(shell pkg-config --define-variable=prefix=$(PREFIX) --silence-errors --variable=completionsdir bash-completion)
+COMPLETIONS_INSTDIR=$(DESTDIR)$(COMPLETIONS_DIR)
+
 libusb_CFLAGS := $(shell $(PKG_CONFIG) --cflags libusb-1.0)
 libusb_LIBS := $(shell $(PKG_CONFIG) --libs libusb-1.0)
 
@@ -95,11 +101,21 @@ install:
 	cp $(MINIPRO) $(BIN_INSTDIR)/
 	cp $(MINIPROHEX) $(BIN_INSTDIR)/
 	cp man/minipro.1 $(MAN_INSTDIR)/
+	if [ -n "$(UDEV_DIR)" ]; then \
+		mkdir -p $(UDEV_RULES_INSTDIR); \
+		cp udev/rules.d/80-minipro.rules $(UDEV_RULES_INSTDIR)/; \
+	fi
+	if [ -n "$(COMPLETIONS_DIR)" ]; then \
+		mkdir -p $(COMPLETIONS_INSTDIR); \
+		cp bash_completion.d/minipro $(COMPLETIONS_INSTDIR)/; \
+	fi
 
 uninstall:
 	rm -f $(BIN_INSTDIR)/$(MINIPRO)
 	rm -f $(BIN_INSTDIR)/$(MINIPROHEX)
 	rm -f $(MAN_INSTDIR)/minipro.1
+	if [ -n "$(UDEV_DIR)" ]; then rm -f $(UDEV_RULES_INSTDIR)/80-minipro.rules; fi
+	if [ -n "$(COMPLETIONS_DIR)" ]; then rm -f $(COMPLETIONS_INSTDIR)/minipro; fi
 
 dist: distclean version-info
 	git archive --format=tar --prefix=minipro-$(GIT_TAG)/ HEAD | tar xf -
-- 
2.20.1

